SPEC.md

```diff
@@ -302,29 +273,89 @@ IEEE 754-2008 decimal floating point numbers
 
 ```kdl
 node (u8)123
-node prop=(regex)".*"
+node prop=(regex).*
 (published)date "1970-01-01"
 (contributor)person name="Foo McBar"
 ```
 
 ### String
 
-Strings in KDL represent textual [Values](#value), or unusual identifiers. A
-String is either a [Quoted String](#quoted-string) or a
-[Raw String](#raw-string). Quoted Strings may include escaped characters, while
-Raw Strings always contain only the literal characters that are present.
+Strings in KDL represent textual UTF-8 [Values](#value). A String is either an
+[Identifier String](#identifier-string) (like `foo`), a [Quoted String](#quoted-string) (like `"foo"`) or
+a [Raw String](#raw-string) (like `#"foo"#`). Identifier Strings let you write short, "single-word" strings with a minimum of syntax; Quoted Strings let you write strings with whitespace (including newlines!) or escapes; Raw Strings let you write strings with whitespace *but without escapes*, allowing you to not worry about the string's content containing anything that might look like an escape.
+
+Strings _MUST_ be represented as UTF-8 values.
+
+Strings _MUST NOT_ include the code points for [disallowed literal code
+points](#disallowed-literal-code-points) directly. Quoted Strings may include
+these code points as _values_ by representing them with their corresponding
+`\u{...}` escape.
+
+### Identifier String
+
+An Identifier String (sometimes referred to as just an "identifier") is
+composed of any [Unicode Scalar
+Value](https://unicode.org/glossary/#unicode_scalar_value) other than
+[non-initial characters](#non-initial-characters), followed by any number of
+Unicode Scalar Values other than [non-identifier
+characters](#non-identifier-characters).
+
+A handful of patterns are disallowed, to avoid confusion with other values:
+
+* idents that appear to start with a [Number](#number) (like `1.0v2` or
+    `-1em`) or the "almost a number" pattern of a decimal point without a
+    leading digit (like `.1`).
+* idents that are the language keywords (`inf`, `-inf`, `nan`, `true`,
+  `false`, and `null`) without their leading `#`.
```

Comment:
I wonder if it would be a good idea to make this a bit more conservative? For example, we could also disallow `NaN`, `-nan`, `+inf`, `Infinity`, etc.